File: Solution/Java/YashG/TflQbMproject/src/main/java/com/tap/yash/entity/Question.java
package com.tap.yash.entity;

public class Question {
     

    private String title;
    private int id ;
    private String optionA ;
    private String optionB ;
    private String optionC ;
    private String optionD ;
    private char CorrectAnswer ;
    private String subject ;
    private int evaluationCriteria;

public Question (){
    this.title=null;
    this.id=0;
    this.optionA=null;
    this.optionB=null;
    this.optionC=null;
    this.optionD=null;
    this.CorrectAnswer='\0';
    this.subject=null;
    this.evaluationCriteria=0;

}

public Question (String title , int id , String optionA , String optionB , String optionC , String optionD , char CorrectAnswer , String subject , int evaluationCriteria){

    this.title=null;
    this.id=0;
    this.optionA=optionA;
    this.optionB=optionB;
    this.optionC=optionC;
    this.optionD=optionD;
    this.CorrectAnswer=CorrectAnswer;
    this.subject=subject;
    this.evaluationCriteria=evaluationCriteria;
}

public String gettitle() {
    return title;
}
public void settitle(String title){
    this.title=title;
}

public int getid(){
    return id;
}

public void setid(int id){
    this.id=id;
}

public String getoptionA(){
    return optionA;
}
public void setoptionA(String optionA){
    this.optionA=optionA;
}
public String getoptionB(){
    return optionB;
}
public void setoptionB(String optionB){
    this.optionB=optionB;
}
public String getoptionC(){
    return optionC;
}
public void setoptionC(String optionC){
    this.optionC=optionC;
}
public String getoptionD(){
    return optionD;
}
public void setoptionD(String optionD){
    this.optionD=optionD;
}
public char getCorrectAnswer(){
    return CorrectAnswer;
}
public void setCorrectAnswer(char CorrectAnswer){
    this.CorrectAnswer=CorrectAnswer;
}
public String subject(){
    return subject;
}
public void subject(String subject){
    this.subject=subject;
}
public int getevaluationCriteria(){
    return evaluationCriteria;
}
public void setevaluationCriteria(){
    this.evaluationCriteria=evaluationCriteria;

}
public String toString() {
        return "Question{" + "title='" + title + ", id=" + id + ", optionA='" + optionA +
                ", optionB='" + optionB +
                ", optionC='" + optionC +
                ", optionD='" + optionD +
                ", correctAnswer=" + CorrectAnswer +
                ", subject=" + subject +
                ", evaluationCriteria=" + evaluationCriteria +
                '}';




}






}

File: Solution/others/CPP/Nirjala/FILE_HANDLING/src/Question.cpp
#include "../include/Question.h"
#include <iostream>
#include <sstream>

Question::Question()  {
    id = 0;
    title = "";
    subject = "";
    evaluation_id = 0;
    option_a = "";
    option_b = "";
    option_c = "";
    option_d = "";
    answer_key = 'a';   
}

Question::Question(int id, const string& subject, const string& title,
                   const string& optionA, const string& optionB,
                   const string& optionC, const string& optionD,
                   char answerKey, int evaluationId)
{
    this->id = id;
    this->subject = subject;
    this->title = title;
    this->option_a = optionA;   
    this->option_b = optionB;
    this->option_c = optionC;
    this->option_d = optionD;
    this->answer_key = answerKey;
    this->evaluation_id = evaluationId;
}



int Question::getId() const { return id; }
void Question::setId(int id) { this->id = id; }

string Question::getSubject() const { return subject; }
void Question::setSubject(const string& subject) { this->subject = subject; }

string Question::getTitle() const { return title; }
void Question::setTitle(const string& title) { this->title = title; }

string Question::getOptionA() const { return option_a; }
void Question::setOptionA(const string& option) { option_a = option; }

string Question::getOptionB() const { return option_b; }
void Question::setOptionB(const string& option) { option_b = option; }

string Question::getOptionC() const { return option_c; }
void Question::setOptionC(const string& option) { option_c = option; }

string Question::getOptionD() const { return option_d; }
void Question::setOptionD(const string& option) { option_d = option; }

char Question::getAnswerKey() const { return answer_key; }
void Question::setAnswerKey(char key) { answer_key = key; }

int Question::getEvaluationId() const { return evaluation_id; }
void Question::setEvaluationId(int evalId) { evaluation_id = evalId; }

void Question::display() const {
    cout << "ID: " << id << "\nSubject: " << subject
         << "\nTitle: " << title
         << "\nA: " << option_a << "\nB: " << option_b
         << "\nC: " << option_c << "\nD: " << option_d
         << "\nAnswer: " << answer_key << "\nEvaluation ID: " << evaluation_id << "\n\n";
}

string Question::serialize() const {
    return to_string(id) + "|" + subject + "|" + title + "|" + option_a + "|" +
           option_b + "|" + option_c + "|" + option_d + "|" + answer_key + "|" + to_string(evaluation_id);
}

Question Question::deserialize(const string& line) {
    Question q;
    stringstream ss(line);
    string token;
    getline(ss, token, '|'); q.setId(stoi(token));
    getline(ss, token, '|'); q.setSubject(token);
    getline(ss, token, '|'); q.setTitle(token);
    getline(ss, token, '|'); q.setOptionA(token);
    getline(ss, token, '|'); q.setOptionB(token);
    getline(ss, token, '|'); q.setOptionC(token);
    getline(ss, token, '|'); q.setOptionD(token);
    getline(ss, token, '|'); q.setAnswerKey(token[0]);
    getline(ss, token, '|'); q.setEvaluationId(stoi(token));
    return q;
}

Question::~Question() {}